### PR TITLE
Posts & Pages: Pages context menu part 2

### LINKED ISSUE #21873: Posts & Pages: Update Pages context menu
- Update the context menus to use the new system design, options are separated by dividers

### PR DESCRIPTION
**Pull request by momo-ozawa**
Closes #21873 

## Description
- Adds "set parent", "set as homepage", "set as posts page" actions to the page context menu
- Removes PostCardCell

## How to test
- Go to the posts list screen
- Look at PageMenuViewModel for test scenarios, and verify the expected menu items are displayed
- Test the "set parent", "set as homepage", "set as posts page" actions and verify they work as expected

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/5e938e11-fc4c-4640-bc38-8c6056e13d2e

## Regression Notes
1. Potential unintended areas of impact
pages context menu

2. What I did to test those areas of impact (or what existing automated tests I relied on)
added more tests to PageMenuViewModel

3. What automated tests I added (or what prevented me from doing so)
see above

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by kean** [comment]
Now that these actions are in a separate menu, it looks much cleaner. Maybe we could also collapse this section as these actions are very rarely used.

(Production issue) if you set a sub-page as a homepage, it's not moved to the top of the list. It is moved on wp.com. The sample applies to "Posts Page"

Screenshot 2023-11-01 at 9 29 08 PM

(Production issue) There is no way to restore page to be a regular page again. There is a way to do that on wp.com.

File: WordPress/Classes/ViewRelated/Pages/PageListViewController+Menu.swift [hunk 1]
```diff
@@ -61,6 +61,23 @@ extension PageListViewController: InteractivePostViewDelegate {
         // Not available for pages
     }
 
+    func setParent(for apost: AbstractPost, at indexPath: IndexPath) {
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Pages/PageListViewController+Menu.swift:64]: (nit) This action doesn't really require an index path, so it'll be nice to remove, but it's not blocking merge. I can remove it once I get this PR working https://github.com/wordpress-mobile/WordPress-iOS/pull/21916.

**Review by kean — APPROVED** [review]
Approved, but left a couple of notes.

File: WordPress/Classes/ViewRelated/Pages/PageListViewController+Menu.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Pages/PageListViewController+Menu.swift:64]: That sounds great, thanks!

**Comment by momo-ozawa** [comment]
@kean Thanks for reviewing! I've added your suggestions to https://github.com/wordpress-mobile/WordPress-iOS/issues/21900

**Pull request merged**